src/plugins/query_enhancements/public/query_assist/components/query_assist_bar.tsx

```diff
@@ -67,7 +54,7 @@ export const QueryAssistBar: React.FC<QueryAssistInputProps> = (props) => {
       question: inputRef.current.value,
       index: selectedIndex,
       language: props.dependencies.language,
-      dataSourceId: dataSourceIdRef.current,
+      dataSourceId: services.data.query.dataSet.getDataSet()?.dataSourceRef?.id,
```

Comment:
same here. If you use the stateful dataset here, you can avoid fetchign the index again and causing a potential drift between the selected dataset and when this function is called

Reply:
> avoid fetchign the index again and causing a potential drift between the selected dataset and when this function is called

didn't understand this part, when the function is called `services.data.query.dataSet.getDataSet()` returns the latest selected dataSet. but anyways i'll add a stateful dataset and this part can share the value